verbose: 1

cam:
  cam_to_use: 0 # -1: undetermined, 0: left, 1: right
  cam_sel_time_interval: 100000 # how long time in second is need to operate camera re-selection again.
  cam_sel_min_perc_plant: 0.01 # the min percentage of plant pixel within the entire seg image, used for cam sel
  width: 320
  height: 240
  left:
#    topic: /usb_cam1/image_raw
    topic: /camera/image_raw
#    K: [159.99999300617776, 0.0, 160.0, 0.0, 159.99999300617776, 120.0, 0.0, 0.0, 1.0]
    K: [226.2, 0.0, 172.5, 0.0, 225.7, 109.4, 0.0, 0.0, 1.0]
#    T: [0, 0, -0.65, 0, 0, 0.45378] # t+ypr, cam extrinsic for bird eye view, where map frame is just under the cam
    T: [0, 0, -0.65, 0, 0, 0.45378] # t+ypr, cam extrinsic for bird eye view, where map frame is just under the cam
  right:
#    topic: /right_rgb_cam/image_raw
    topic: /camera/image_raw
    K: [226.2, 0.0, 172.5, 0.0, 225.7, 109.4, 0.0, 0.0, 1.0]
    T: [0, 0, -0.65, 0, 0, 0.45378] # t+ypr, cam extrinsic for bird eye view, where map frame is just under the cam

lane_det:
  yolo_plant_id: 1
  yolo_model: best_offline21.pt
  wrap_img_on: 1
  # wrap_param: [180, 319, 200, 0, 239, 250] # [left lane bot pix u, right lane bot pix u, mid lane top pix u, top v, bot v, weeder bot u]
  # wrap_param: [120, 225, 180, 0, 220, 250] # [left lane bot pix u, right lane bot pix u, mid lane top pix u, top v, bot v, weeder bot u]
  wrap_param_left_cam: [128, 241, 160, 189, 239,   0, 184]  # [left lane bot pix u, right lane bot pix u, left lane top pix u, right lane top pix u, bot v, top v, weeder bot u]
  wrap_param_right_cam: [139, 226, 156, 197, 188,  41, 182] # [left lane bot pix u, right lane bot pix u, left lane top pix u, right lane top pix u, bot v, top v, weeder bot u]
  wrap_img_size_u: 100
  track_on: 1
  track_line_width: 1
  track_line_filter_width: 7
  track_img_pre_weight: 1
  theta_range: 5 # search left line's theta angle from [-theta_range + ref_theta, theta_range + ref_theta]
  shift_range: 5 # search right line's bottom u coords from [-shift_range + left_line_bot_u + ref_lane_width, shift_range + left_line_bot_u + ref_lane_width]
  track_u_width: 5

weeder:
  farm_lane_dist_left_cam: 0.4
  farm_lane_dist_right_cam: 0.4
  def_speed: 0.5 # default weeder speed, m/s
  def_pos: 0.0
  cam_weeder_dist: 1.6
  cmd_dist: -1 # 0.1 # the minimum distance the weeder should travel for the new control cmd to be sent
  cmd_dist_def: 0.1 # the minimum distance the weeder should travel for the new control cmd to be sent
  ser_port: /dev/ttyS0 # serial port for weeder mic
  baudrate: 115200
  bytesize: 8  # len
  stopbits: 1
  parity: N
  mic_response_time: 0.1 # after weeder control cmd is send, how long it should wait for weeder mic to respond
  hardware_interaction: 1 # whether weeder mic is connected, enable serial, otherwise not
  verbose: 1
  weeder_cmd_delay: 0
  weeder_cmd_max_shift: 0.05

log:
  enable: 1

tf:
  Tlwlc: [1.619993, -0.000001, 0.650002, -0.653281, 0.653281, -0.270598, 0.270598] # t+quat, left cam in left weeder
  Tblw: [0.000004, 2.250000, -0.000002, 0.000000, 0.000000, 0.000000, 1.000000] # t+quat, left weeder in base_link
  Trwrc: [1.620018, 0.000000, 0.650000, -0.653282, 0.653282, -0.270598, 0.270598] # t+quat, right cam in left weeder
  Tbrw: [0.000002, -2.249999, 0.000000, 0.000000, 0.000000, 0.000000, 1.000000] # t+quat, right weeder in base_link




